plugins/flow-visibility/clickhouse-monitor/main.go

```diff
@@ -0,0 +1,256 @@
+	// I0207 22:29:06.283450       1 main.go:71] "Next round will not be skipped" skipRoundsNum=0
+	// ...
+
+	// reads the number of rounds requires to be skipped
+	logs := strings.Split(logString, "skipRoundsNum=")
+	if len(logs) != 2 {
+		klog.ErrorS(nil, "Error when finding number of rounds")
+		return false
+	}
+	lines := strings.Split(logs[1], "\n")
+	remainingRoundsNum, convErr := strconv.Atoi(lines[0])
+	if convErr != nil {
+		klog.ErrorS(convErr, "Error when finding last monitor job")
+		return false
+	}
+	if remainingRoundsNum > 0 {
+		klog.InfoS("Skip rounds after a successful deletion", "skipRoundsNum", remainingRoundsNum-1)
+		return true
+	}
+	return false
+}
+
+// Gets pod logs from the Clickhouse monitor job
+func getPodLogs() (string, error) {
+	var logString string
+	podLogOpts := corev1.PodLogOptions{}
+	config, err := rest.InClusterConfig()
+	listOptions := metav1.ListOptions{
+		LabelSelector: monitorLabel,
+	}
+	if err != nil {
+		return logString, fmt.Errorf("error when getting config: %v", err)
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return logString, fmt.Errorf("error when getting access to K8S: %v", err)
+	}
+	// gets Clickhouse monitor pod
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return logString, fmt.Errorf("failed to list clickhouse monitor Pods: %v", err)
+	}
+	for _, pod := range pods.Items {
+		// reads logs from the last successful pod
+		if pod.Status.Phase == corev1.PodSucceeded {
+			req := clientset.CoreV1().Pods(namespace).GetLogs(pod.Name, &podLogOpts)
+			podLogs, err := req.Stream(context.TODO())
+			if err != nil {
+				return logString, fmt.Errorf("error when opening stream: %v", err)
+			}
+			defer podLogs.Close()
+
+			buf := new(bytes.Buffer)
+			_, err = io.Copy(buf, podLogs)
+			if err != nil {
+				return logString, fmt.Errorf("error when copying information from podLogs to buf: %v", err)
+			}
+			logString := buf.String()
+			return logString, nil
+		}
+	}
+	return logString, fmt.Errorf("no successful monitor")
+}
+
+// Connects to Clickhouse in a loop
+func connectLoop() (*sql.DB, error) {
+	// Clickhouse configuration
+	userName := os.Getenv("CH_USERNAME")
+	password := os.Getenv("CH_PASSWORD")
+	host, port := os.Getenv("SVC_HOST"), os.Getenv("SVC_PORT")
+
+	ticker := time.NewTicker(connectionWait)
+	defer ticker.Stop()
+
+	timeoutExceeded := time.After(connectionTimeout)
+	for {
+		select {
+		case <-timeoutExceeded:
+			return nil, fmt.Errorf("failed to connect to Clickhouse after %s", connectionTimeout)
+
+		case <-ticker.C:
+			// Open the database and ping it
+			dataSourceName := fmt.Sprintf("tcp://%s:%s?debug=true&username=%s&password=%s", host, port, userName, password)
+			connect, err := sql.Open("clickhouse", dataSourceName)
+			if err != nil {
+				klog.ErrorS(err, "Failed to connect to Clickhouse")
+			}
+			if err := connect.Ping(); err != nil {
```

Comment:
just a curiosity: is the ping operation necessary to assess the connection is healthy?

Reply:
This is suggested by the [clickhouse-go](https://github.com/ClickHouse/clickhouse-go/tree/v1) examples.